### PR TITLE
Port MYFACES-4432

### LINKED ISSUE #21601: Port MYFACES-4432 to JSF 2.3 and Faces 3.0 (Resolve request object in facelets)
**Describe the bug**  
When using the "@FacesConfig" annotation in a JSF 2.3 or Faces 3.0 application, the _#{request}_ object fails to resolve. An empty string is returned instead. 

**Steps to Reproduce**  
- Use @FacesConfig in the 2.3 or 3.0 application, and reference _request_  in a facelet. 

**Expected behavior**  
The object should resolve, so that any of it's methods can be called. See table 5-11  of the JSF 2.3 spec for all objects which should be resolved. 

**Diagnostic information:**  
 - OpenLiberty Version: All versions
 - Affected feature(s) jsf-2.3, faces-3.0
 - Java Version: N/A

**Additional context**  
Note: The implicit request and session objects are resolved via JSF's _ImplicitObjectResolver_ not CDI. This is so that HttpServletRequest and HttpSession are not made injectable by JSF (i.e.. JSF should not use the Producer annotation in the implementation for HttpServletRequest and HttpSession). CDI should directly manage the injectablility of these objects instead. See the the discussion  `https://github.com/jakartaee/faces/issues/1564` for reference. 
https://issues.apache.org/jira/projects/MYFACES/issues/MYFACES-4432

### PR DESCRIPTION
**Pull request by volosied**
fixes #21601

### CODE REVIEW

**Comment by volosied** [comment]
#build

**Comment by wraschke** [comment]
#libby

**Comment by wraschke** [comment]
Don't pay attention to me.

File: dev/com.ibm.ws.jsf.2.3_fat/fat/src/com/ibm/ws/jsf23/fat/tests/JSF23CDIGeneralTests.java [hunk 1]
```diff
@@ -324,6 +324,7 @@ private void checkInjectableELImplicitObjects(WebClient webClient) throws Except
         assertTrue(resultPage.asText().contains("Message from RequestParameterMap: Hello World"));
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.3_fat/fat/src/com/ibm/ws/jsf23/fat/tests/JSF23CDIGeneralTests.java:324]: Update the copyright date range?

File: dev/com.ibm.ws.jsf.2.3_fat/test-applications/ELImplicitObjectsViaCDI.war/resources/implicit_objects.xhtml [hunk 2]
```diff
@@ -63,5 +63,11 @@
         <br/>
         <br/>
         ViewScope isEmpty: <h:outputText value="#{viewScope.isEmpty()}"/>
+        <br/>
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.3_fat/test-applications/ELImplicitObjectsViaCDI.war/resources/implicit_objects.xhtml:66]: Update copyright date range?

File: dev/com.ibm.ws.jsf.2.3_fat/test-applications/ELImplicitObjectsViaCDI.war/resources/implicit_objects.xhtml [hunk 3]
```diff
@@ -63,5 +63,11 @@
...
+        <br/>
+        Request contextPath: <h:outputText value="#{request.contextPath}"/>
+        <br/>
+        <br/>
+        ExternalContext getApplicationContextPath: <h:outputText value="#{externalContext.getApplicationContextPath()}"/>
     </h:body>
 </html>
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.3_fat/test-applications/ELImplicitObjectsViaCDI.war/resources/implicit_objects.xhtml]: newline please.

File: dev/com.ibm.ws.jsf.2.3_fat/fat/src/com/ibm/ws/jsf23/fat/tests/JSF23CDIGeneralTests.java [hunk 4]
```diff
@@ -379,9 +380,15 @@ public void testELResolutionImplicitObjects() throws Exception {
             assertTrue(testELResolutionImplicitObjectsPage.asText().contains("InitParam: ELImplicitObjectsViaCDI"));
             assertTrue(testELResolutionImplicitObjectsPage.asText().contains("Param: Hello World"));
             assertTrue(testELResolutionImplicitObjectsPage.asText().contains("ParamValues: Hello World"));
-            assertTrue(testELResolutionImplicitObjectsPage.asText().contains("Session isNew: true"));
             assertTrue(testELResolutionImplicitObjectsPage.asText().contains("View viewId: /implicit_objects.xhtml "));
             assertTrue(testELResolutionImplicitObjectsPage.asText().contains("ViewScope isEmpty: true"));
+            // See https://issues.apache.org/jira/projects/MYFACES/issues/MYFACES-4432
+            // Note: The request & session objects are not resolved by CDI, but via ImplicitObjectResolver
+            assertTrue(testELResolutionImplicitObjectsPage.asText().contains("Request contextPath: /ELImplicitObjectsViaCDI"));
+            assertTrue(testELResolutionImplicitObjectsPage.asText().contains("Session isNew: true"));
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.3_fat/fat/src/com/ibm/ws/jsf23/fat/tests/JSF23CDIGeneralTests.java]: I believe the actual output is `HttpSession isNew:` can we be more specific?

File: dev/com.ibm.ws.jsf.2.3_fat/test-applications/ELImplicitObjectsViaCDI.war/src/com/ibm/ws/jsf23/fat/elimplicit/cdi/beans/ELImplicitObjectBean.java [hunk 5]
```diff
@@ -39,6 +39,7 @@
 import javax.inject.Named;
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.3_fat/test-applications/ELImplicitObjectsViaCDI.war/src/com/ibm/ws/jsf23/fat/elimplicit/cdi/beans/ELImplicitObjectBean.java:39]: Update copyright date?

File: dev/com.ibm.ws.jsf.2.3_fat/fat/src/com/ibm/ws/jsf23/fat/tests/JSF23CDIGeneralTests.java (hunk 4 above)
Comment (volosied) [reply dev/com.ibm.ws.jsf.2.3_fat/fat/src/com/ibm/ws/jsf23/fat/tests/JSF23CDIGeneralTests.java]: Fixed

**Review by pnicolucci — APPROVED** [review]

**Comment by volosied** [comment]
#libby

**Pull request merged**